wire/msgblock.go

```diff
@@ -137,23 +137,28 @@ func (msg *MsgBlock) DeserializeNoWitness(r io.Reader) error {
 func (msg *MsgBlock) DeserializeTxLoc(r *bytes.Buffer) ([]TxLoc, error) {
 	fullLen := r.Len()
 
+	buf := binarySerializer.Borrow()
```

Comment:
Not sure if it's helpful to mention every instance where we can use `defer binarySerializer.Return()` to mark all the commits/files that need to be updated. I'm guessing not, so I'm not going to mention it any further.
It does sound like a tedious update in a lot of commits, but IMO it's worth it to avoid the buffer channel to fill up and effectively become useless because of a missing `Return()` somewhere.

Reply:
Good point, i think when this was originally written there might have been some concerns w.r.t the performance of `defer`. However since then, things have been optimized significantly.